src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -167,8 +167,15 @@ CHIP_ERROR DeviceControllerFactory::InitSystemState(FactoryInitParams params)
     ReturnErrorOnFailure(stateParams.transportMgr->Init(Transport::UdpListenParameters(stateParams.udpEndPointManager)
                                                             .SetAddressType(Inet::IPAddressType::kIPv6)
                                                             .SetListenPort(params.listenPort)
-#if INET_CONFIG_ENABLE_IPV4
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
                                                             ,
+                                                        Transport::TcpListenParameters(stateParams.tcpEndPointManager)
+                                                            .SetAddressType(IPAddressType::kIPv6)
+                                                            .SetListenPort(params.listenPort)
+                                                            .SetServerListenEnabled(false) // Initialize as a TCP Client
```

Comment:
Wait, why would we force controllers to never be TCP servers?  This isn't a "by default" like the PR description misleadingly says: it's just force-disabled.

Reply:
In order to keep the interaction simple, I was assuming controllers(like chip-tool and commissioning clients) to be TCP clients. 
The TcpListenParameters probably need to be percolated further up in order to make it a truly configurable parameter. For now, I have set the controller to be a TCP Client.